core/src/main/java/org/mule/runtime/core/internal/policy/DefaultPolicyInstance.java

```diff
@@ -77,18 +73,22 @@
 
   private ProcessingStrategy processingStrategy;
 
+  private String name;
+
   private PolicyChain operationPolicyChain;
   private PolicyChain sourcePolicyChain;
 
-  private FlowConstructStatistics flowConstructStatistics = new DefaultFlowConstructStatistics("policy", getName());
-  private String name = "proxy-policy-" + UUID.getUUID();
+  private FlowConstructStatistics flowConstructStatistics;
   private MuleContext muleContext;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "MuleContext"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7629&issues=AWiGlnXCotIXnmsUoniX&open=AWiGlnXCotIXnmsUoniX)

[ISSUE_KEY]: AWiGlnXCotIXnmsUoniX